### ISSUE TITLE
QueryGenerator or QueryConstructor

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#1168]
Here: https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SearchCollection.java#L127

We're inconsistent in the use of `QueryGenerator` and `QueryConstructor`: should we standardize? I think I like `QueryConstructor` better, so this would entail renaming all the classes?

Thoughts @Chriskamphuis ?

**Comment by Chriskamphuis** [comment]
I agree, especially if we ship this to Python it can be confusing with the native python generator objects.

**Comment by lintool** [comment]
The method name is called `buildQuery`... what about "builder"?

So, builder vs. constructor vs. generator? Maybe builder is even better because it doesn't get confused with class constructors?

**Comment by Chriskamphuis** [comment]
Sure, however the native Lucene Query objects have a Builder object they use. So it would look something like this:
```
PhraseQueryBuilder:
-- BuildQuery()
----PhraseQuery.Builder builder = new PhraseQuery.Builder()
------builder.build()
```

which is a bit confusing maybe?

**Comment by lintool** [comment]
hrm. I like calling it "builder" but if we don't adopt Lucene's idiom, will we create confusion? I think separate builder classes in our context is over-engineering...

**Comment by lintool** [comment]
Clarified over Slack. @Chriskamphuis 's main point above is that if we named our class Builder, then internally we'll have to call Lucene's builder, which makes it terrible confusing with variable names... 

tl;dr - since all the terms are overloaded and ambiguous (but in different ways), we're just going to stick to "generator", so the fix is to rename the "constructor" in `SearchCollection`.
